axelrod/strategies/_filters.py

```diff
@@ -0,0 +1,179 @@
+            equal to' test should be applied.
+
+    Returns
+    -------
+        boolean
+
+        True if the value from the strategy's classifier dictionary matches
+        the value and operator passed to the function.
+    """
+    if isinstance(value, str):
+        filter_value = int(value)
+    else:
+        filter_value = value
+
+    classifier_value = strategy.classifier[classifier_key]
+    if (isinstance(classifier_value, str) and
+            classifier_value.lower() == 'infinity'):
+        classifier_value = float('inf')
+
+    return operator(classifier_value, filter_value)
+
+
+def passes_in_list_filter(strategy, classifier_key, value):
+    """
+    Tests whether a given value exists in the list returned from the given
+    strategy's classifier dict for the given classifier_key.
+
+    Parameters
+    ----------
+        strategy : a descendant class of axelrod.Player
+        classifier_key: string
+            Defining which entry from the strategy's classifier dict is to be
+            tested (e.g. 'makes_use_of').
+        value: string
+            The value against which the strategy's classifier dict entry is to
+            be tested.
+
+    Returns
+    -------
+        boolean
+    """
+    return value in strategy.classifier[classifier_key]
+
+
+def passes_filterset(strategy, filterset):
+    """
+    Determines whether a given strategy meets the criteria defined in a
+    dictionary of filters.
+
+    Parameters
+    ----------
+        strategy : a descendant class of axelrod.Player
+        filterset : dict
+            mapping filter name to criterion.
+            e.g.
+                {
+                    'stochastic': True,
+                    'min_memory_depth': 2
+                }
+
+    Returns
+    -------
+        boolean
+
+        True if the given strategy meets all the supplied criteria in the
+        filterset, otherwise false.
+
+    """
+    FilterFunction = namedtuple('FilterFunction', 'function kwargs')
+
+    # A dictionary mapping filter name (from the supplied filterset) to
+    # the relevant function and arguments for that filter.
+    filter_functions = {
+        'stochastic': FilterFunction(
+            function=passes_boolean_filter,
+            kwargs={'classifier_key': 'stochastic'}),
+        'long_run_time': FilterFunction(
+            function=passes_boolean_filter,
+            kwargs={'classifier_key': 'long_run_time'}),
+        'manipulates_state': FilterFunction(
+            function=passes_boolean_filter,
+            kwargs={'classifier_key': 'manipulates_state'}),
+        'manipulates_source': FilterFunction(
+            function=passes_boolean_filter,
+            kwargs={'classifier_key': 'manipulates_source'}),
+        'inspects_source': FilterFunction(
+            function=passes_boolean_filter,
+            kwargs={'classifier_key': 'inspects_source'}),
+        'min_memory_depth': FilterFunction(
```

Comment:
I think I've seen why you've done this so my question might not make terrible sense but could we perhaps also include `memory_depth` in here? So that it picks out exactly the strategies with a given memory depth? I'm not sure my suggestion is a good one, I just like the idea of the player classifier keys having the same keys (as much as possible) as this filter key... (I'm thinking out loud here)